BLUESPAWN-win-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -1,294 +1,298 @@
+                            return std::to_wstring(result->UInt32Val);
+                        } else if(result->Type == EvtVarTypeUInt64) {
+                            return std::to_wstring(result->UInt64Val);
+                        } else if(result->Type == EvtVarTypeNull)
+                            return L"NULL";
+                        else {
+                            return L"Unknown VARIANT: " + std::to_wstring(result->Type);
+                        }
+                    }
+                }
+            }
+        }
+        return std::nullopt;
+    }
+
+    std::optional<std::wstring> EventLogs::GetEventXML(const EventWrapper& hEvent) {
+        DWORD dwBufferSize = 0;
+        if(!EvtRender(NULL, hEvent, EvtRenderEventXml, dwBufferSize, nullptr, &dwBufferSize, nullptr)) {
+            if(ERROR_INSUFFICIENT_BUFFER == GetLastError()) {
+                auto pRenderedContent = AllocationWrapper{ HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, dwBufferSize),
+                                                           dwBufferSize, AllocationWrapper::HEAP_ALLOC };
+                if(pRenderedContent) {
+                    if(EvtRender(NULL, hEvent, EvtRenderEventXml, dwBufferSize, pRenderedContent, &dwBufferSize,
+                                 nullptr)) {
+                        return reinterpret_cast<LPCWSTR>((LPVOID) pRenderedContent);
+                    }
+                }
+            }
+        }
+
+        return std::nullopt;
+    }
+
+    // Enumerate all the events in the result set.
+    std::vector<EventLogItem> EventLogs::ProcessResults(const EventWrapper& hResults,
+                                                        const std::vector<XpathQuery>& filters) {
+        EVT_HANDLE hEvents[ARRAY_SIZE];
+
+        std::vector<EventLogItem> results;
+        std::vector<std::wstring> params;
+        for(auto query : filters) {
+            if(!query.SearchesByValue()) {
+                params.push_back(query.ToString());
+            }
+        }
+
+        DWORD dwReturned{};
+        while(EvtNext(hResults, ARRAY_SIZE, hEvents, INFINITE, 0, &dwReturned)) {
+            for(DWORD i = 0; i < dwReturned; i++) {
+                auto item = EventToEventLogItem(hEvents[i], params);
+                if(item) {
+                    results.push_back(*item);
+                }
+
+                EvtClose(hEvents[i]);
+                hEvents[i] = NULL;
+            }
+        }
+
+        for(unsigned i = 0; i < ARRAY_SIZE; i++) {
+            if(hEvents[i]) {
+                EvtClose(hEvents[i]);
+            }
+        }
+
+        if(GetLastError() != ERROR_NO_MORE_ITEMS) {
+            LOG_VERBOSE(1, "EventLogs::ProcessResults: EvtNext failed with " << GetLastError());
+        }
+
+        return results;
+    }
+
+    std::optional<EventLogItem> EventToEventLogItem(const EventWrapper& hEvent,
+                                                    const std::vector<std::wstring>& params) {
```

Comment:
Codacy found an issue: [Variable 'status' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5931486)